x/supplier/client/cli/stake_options_reader.go

```diff
@@ -0,0 +1,157 @@
+package cli
```

Comment:
I think this is a great step forward, but believe it needs two things:

1. Unit tests specific to the reader
2. A README, because all node runners are going to reference supplier configs

If you think it should be done in followup PRs, instead of this one, please add the appropriate TODO_BLOCKER/TODO_TECHDEBT comments. Personally, I think unit tests should be part of this PR, and the documentation should be a TODO_BLOCKER.